### PR TITLE
feat: #12 add enable flag

### PR DESCRIPTION
**Pull request by baxyz**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #23

## What is the new behavior?

This PR adds a simple flag (`enable`) to allow or forbid edition mode. This flag doesn't swich to edition mode, only prevent if disabled.

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by geromegrignon** [comment]
Hey @baxyz, I'll review your PR tomorrow.

**Review by geromegrignon — CHANGES_REQUESTED** [review]
Can you change `enable` to `enabled`?
I prefer to use passive voice for properties, keeping active voice for events.

File: src/app/app.component.html [hunk 1]
```diff
@@ -401,4 +401,43 @@ <h4>Grouped inputs</h4>
...
+    <input
+      #enableFlagInput
+      class="form-check-input"
+      type="checkbox"
+      id="enable-flag"
+      name="enable-flag"
+      checked
+      data-testid="sample-enable-flag-check"
+    />
+    <label for="enable-flag" class="form-check-label">Allows edition</label>
```
Comment (geromegrignon) [inline src/app/app.component.html:416]: please change `edition` to `edit`

**Review by geromegrignon — CHANGES_REQUESTED** [review]

File: src/app/app.component.html (hunk 1 above)
Comment (baxyz) [reply src/app/app.component.html]: done

**Comment by baxyz** [comment]
> Can you change `enable` to `enabled`? I prefer to use passive voice for properties, keeping active voice for events.

done

**Comment by geromegrignon** [comment]
> > Can you change `enable` to `enabled`? I prefer to use passive voice for properties, keeping active voice for events.
> 
> done

The change must be done in the README and the test too ( currently failing).

**Comment by baxyz** [comment]
> > > Can you change `enable` to `enabled`? I prefer to use passive voice for properties, keeping active voice for events.
> > 
> > 
> > done
> 
> The change must be done in the README and the test too ( currently failing).

Done.
I forgot some files during the fixup.

File: README.md [hunk 2]
```diff
@@ -197,6 +197,7 @@ The following actions will trigger this event:
 | ---------------------- | ------------------------- | ------------------------------------------------------------ | ---------------------------------------------------------------------- |
 | openBindingEvent       | `string`                  | The MouseEvent type to display the editMode                  | `click`                                                              |
 | closeBindingEvent      | `string`                  | The MouseEvent type to display the viewMode                  | `click`                                                                 |
+| enabled                | `boolean`                 | Allows or forbids edit mode (doesn't swith to it)            | `true`                                                                 |
```
Comment (geromegrignon) [inline README.md:200]: sorry, one last change as there is a typo: from `swith` to `switch`

Comment (baxyz) [reply README.md]: 👍 right

**Comment by geromegrignon** [comment]
Thanks @baxyz for your contribution.
I have a few tasks left to update the project to Angular 13 and I'll release a new version.

**Pull request merged**